sapirbitan/assignment2/ex2.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+using namespace std;
+
+struct node {
+	int key;
+	node* left_child = NULL;
+	node* right_child = NULL;
+	node* ancestor = NULL;
+};//assuming every node contains pointers to it's children and ancestors
+  //(the root's ancestor will point to NULL)
+
+/****************************************************************************************
+This function accepts three nodes - the root of the tree, and the two nodes we want to find
+the common ancestor of.
+how does she do it?
+It goes over all of the second node's ancestor and compares them with all of the first node's 
+ancestors.
+Worst case scenario - both of the nodes are at the end of the tree - which meens we'll have 
+to compare log(n) [n is num of nodes in the tree] ancestors with log(n) ancestors - les then o(n)
+*****************************************************************************************/
+node* find_common_anc(const node& tree_root,const node& first_node, const node& second_node) {
+	//Since I don't want to change the original nodes(ans can't - because they're const)
+	//I am using temporary nodes which I can change
+	node const_second = second_node;
+	node temp_first = first_node;
+	node temp_second = second_node;
+	node temp_tree_root = tree_root;
+	int flag=1;//checks if we've found the common ancestor
```

Comment:
I would name the flag something more meaningful, like `not_found` :) also it should probably be a boolean:

`bool not_found = true; // checks if we've found the common ancestor`